### ISSUE TITLE
Making psych output array in flow format is harder that it should be

### ISSUE THREAD

**Issue by hkmaly** [ruby/psych#281]
Psych allows to parametrize to_yaml output by defining method encode_with on object. But this parametrization is sparely documented and is not as universal as it could be. For example, to force all arrays to display in FLOW style, following might work:

```
class Array
  def encode_with(coder)
    coder.style = Psych::Nodes::Mapping::FLOW
    coder.tag = nil
    coder.seq = self
  end
end
```

(Obviously, this is just easy to test example: in real use, there would either be some if or it wouldn't be Array directly).

But this simple modification doesn't work, because in psych/visitors/yaml_tree.rb inside function emit_coder the coder.style is ignored for seqences by this lines:

```
        when :seq
          @emitter.start_sequence nil, c.tag, c.tag.nil?, Nodes::Sequence::BLOCK
```

Why couldn't it be 

```
        when :seq
          @emitter.start_sequence nil, c.tag, c.tag.nil?, c.style
```

instead?

**Comment by tenderlove** [comment]
I think it could be. Can you write a a patch?

**Comment by hkmaly** [comment]
You mean like this?
```
diff -Naur 1/yaml_tree.rb 2/yaml_tree.rb
--- 1/yaml_tree.rb      2016-04-28 23:12:40.000000000 +0200
+++ 2/yaml_tree.rb      2016-04-28 23:13:45.000000000 +0200
@@ -568,7 +568,7 @@
         when :scalar
           @emitter.scalar c.scalar, nil, c.tag, c.tag.nil?, false, Nodes::Scalar::ANY
         when :seq
-          @emitter.start_sequence nil, c.tag, c.tag.nil?, Nodes::Sequence::BLOCK
+          @emitter.start_sequence nil, c.tag, c.tag.nil?, c.style
           c.seq.each do |thing|
             accept thing
           end
```

**Comment by hkmaly** [comment]
Actually, while at this, maybe the scalar (two lines above the seq) should be changed as well - but that might require to set some default.

**Comment by hkmaly** [comment]
Any progress on this?

**Comment by sandstrom** [comment]
friendly ping @tenderlove 😄

**Comment by tenderlove** [comment]
Yes, please send a PR. Thanks!

**Comment by hkmaly** [comment]
Ok, I tried.

**Comment by hkmaly** [comment]
@tenderlove I suppose we can close this as you solved it in #329 good enough?
